src/react.tsx

```diff
@@ -15,23 +38,36 @@ export function connect<StoreState extends object>(store: StoreDefinition<StoreS
   ): React.ComponentClass<Diff<PropsWithStore, { store: Store<StoreState> }>> {
 
     type State = {
-      store: StoreSnapshot<StoreState>
-      subscription: Subscription
+      store: StoreSnapshotWrapper<StoreState>
     }
 
     return class extends React.Component<Diff<PropsWithStore, { store: Store<StoreState> }>, State> {
       static displayName = `withStore(${getDisplayName(Component)})`
+      const subscriptions: Map<keyof StoreState, Subscription> = new Map()
+
+      const _onGet = (field: keyof StoreState) => {
```

Comment:
Any reason this shouldn't be a method this class?

Reply:
I instantiate it as part of the new React.Component, but it is a method. Could you please elaborate?

Reply:
I see - I was confused by the `const` keyword. I'm not sure why TypeScript isn't complaining there. What you want instead is:

```diff
-const _onGet = (field: keyof StoreState) => {
+private onGet = (field: keyof StoreState) => {
```

Reply:
Looks like this was a TS bug - `const` indeed is not allowed as a member variable modifier. https://github.com/Microsoft/TypeScript/issues/25123

Reply:
I have no idea about TS. I am too used to Flow.